pkg/cmd/browse/browse.go

```diff
@@ -0,0 +1,245 @@
+package browse
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type browser interface {
+	Browse(string) error
+}
+
+type BrowseOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Browser    browser
+
+	SelectorArg   string
+	AdditionalArg string
+
+	ProjectsFlag bool
+	WikiFlag     bool
+	SettingsFlag bool
+	BranchFlag   bool
+}
+
+type exitCode int
+
+const (
+	exitUrlSuccess    exitCode = 0
+	exitNonUrlSuccess exitCode = 1
+	exitNotInRepo     exitCode = 2
+	exitTooManyFlags  exitCode = 3
+	exitTooManyArgs   exitCode = 4
+	exitExpectedArg   exitCode = 5
+	exitInvalidCombo  exitCode = 6
+)
+
+func NewCmdBrowse(f *cmdutil.Factory) *cobra.Command {
+
+	opts := &BrowseOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Browser:    f.Browser,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Long:  "Open specific pages in the browser on GitHub",
+		Short: "Open GitHub in the browser",
+		Use:   "browse {<number> | <path>}",
+		Args:  cobra.RangeArgs(0, 2),
+		Example: heredoc.Doc(`
+			$ gh browse
+			#=> Open current repository to the default branch
```

Comment:
How about:
```suggestion
			#=> Open the home page of the current repository
```
It's more relevant here that the main or home page of the repository will open. The fact that this will also be on the default branch should be understood by most people who have already used GitHub from their web browser.